### PR TITLE
ovl_Demo_6K

### PR DESCRIPTION
**Pull request by maekclena**
func_80968B70 is non-matching but likely equivalent.
func_80967FFC and func_809688C4 have probable fake matches.

### CODE REVIEW

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 1]
```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
     (ActorFunc)Demo6K_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_6K/func_80966950.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_6K/Demo6K_Init.s")
+static s16 sObjectIds[] = {
+    1, OBJECT_DEMO_6K, OBJECT_DEMO_6K, 1, 1, 1, 1, 1, 1, 1, 1, 1, OBJECT_GND_MAGIC, 1, 1, 1, 1, 1, 1, 1
```
Comment (petrie911) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:58]: Object 1 is `OBJECT_GAMEPLAY_KEEP`

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 2]
```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
...
+void Demo6K_Init(Actor* thisx, GlobalContext* globalCtx) {
+    Demo6K* this = THIS;
+    s32 pad;
+    s32 params = this->actor.params;
+    s32 objBankIndex;
+    s32 i;
+
+    osSyncPrintf("no = %d\n", params);
+
+    if (sObjectIds[params] != 1) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:79]:
```suggestion
    if (sObjectIds[params] != OBJECT_GAMEPLAY_KEEP) {
```

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 3]
```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
...
+        this->unk_168 = D_8096930C[this->timer1 & 3];
+    } else if (this->timer1 < 15) {
+        this->actor.scale.x += 0.012f;
+        Actor_SetScale(&this->actor, this->actor.scale.x);
+        Math_StepToF(&this->unk_170, 0.6f, 0.05f);
+        this->unk_168 = 1.0f;
+    } else {
+        if (this->timer1 == 15) {
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EFF_DUST, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 1);
```
Comment (petrie911) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:259]:
```suggestion
                               this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, EFF_DUST_TYPE_1);
```

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 4]
```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
...
+        this->timer1++;
+    } else if (this->actor.scale.x > 0.0f) {
+        this->actor.scale.x -= 1.0f / 120.0f;
+        Actor_SetScale(&this->actor, this->actor.scale.x);
+    } else {
+        Actor_Kill(&this->actor);
+    }
+}
+
+static u16 D_8096932C[] = { 275, 275, 275, 275, 275, 275 };
```
Comment (petrie911) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:281]: This goes inside `func_8096712C`?

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 5]
```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
...
+
+    this->timer2++;
+
+    if ((globalCtx->sceneNum == SCENE_GANONTIKA) && (globalCtx->csCtx.frames < D_8096932C[this->actor.params - 3])) {
+        func_8002F974(&this->actor, NA_SE_EV_LIGHT_GATHER - SFX_FLAG);
+    }
+}
+
+void func_80967244(Demo6K* this, GlobalContext* globalCtx) {
+    static Vec3f sVelocity = { 0.0f, 0.0f, 0.0f };
```
Comment (petrie911) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:311]: in-function static variables don't need to be prefixed with `s`

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 6]
```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
...
+    gDPSetEnvColor(POLY_XLU_DISP++, envColor[0], envColor[1], envColor[2], 128);
+    gSPDisplayList(POLY_XLU_DISP++, D_06001190);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_demo_6k.c", 1368);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_6K/func_80968B70.s")
+#endif
+
+static u8 D_809693CC[] = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 7, 6, 5, 4, 3, 2, 1 };
```
Comment (petrie911) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:782]: Goes inside `func_8096712C`?

**Review by petrie911 — COMMENTED** [review]
Looks pretty good. Just a few things.

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 7]
```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
...
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_demo_6k.c", 1115),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gDPPipeSync(POLY_XLU_DISP++);
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, colors[i][0].r, colors[i][0].g, colors[i][0].b, 255);
+            gDPSetEnvColor(POLY_XLU_DISP++, colors[i][1].r, colors[i][1].g, colors[i][1].b, 255);
+            gSPDisplayList(POLY_XLU_DISP++, D_060022B0);
+        }
+
+        // required to avoid optimizing out i
+        if ((s16)i) {}
```
Comment (ProjectRevoTPP) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:595]: does IDO fail to optimize out a second for loop here with no body? Does that have the same effect?

Comment (maekclena) [reply src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:595]: No, I can't get it to match any other way.

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_Demo_6K/z_demo_6k.c [hunk 8]
```diff
@@ -27,57 +54,781 @@ const ActorInit Demo_6K_InitVars = {
...
+
+    this->timer2++;
+
+    Actor_SetScale(&this->actor, 0.2f);
+
+    if ((globalCtx->csCtx.state != CS_STATE_IDLE) && (globalCtx->csCtx.npcActions[params] != NULL)) {
+        func_809691BC(this, globalCtx, params);
+
+        if (globalCtx->csCtx.npcActions[params]->action == 3) {
+            this->flags &= 0xFFFE;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:362]:
```suggestion
            this->flags &= ~1;
```

Comment (fig02) [reply src/overlays/actors/ovl_Demo_6K/z_demo_6k.c:362]: gonna go ahead and commit this directly since its such a small change and the only one

**Review by Roman971 — APPROVED** [review]

**Pull request merged**